main/src/test/java/org/mobilitydata/gtfsvalidator/util/GeospatialUtilTest.java

```diff
@@ -0,0 +1,650 @@
+        .setCsvRowNumber(csvRowNumber)
+        .setShapeId(shapeId)
+        .setShapePtLat(shapePtLat)
+        .setShapePtLon(shapePtLon)
+        .setShapePtSequence(shapePtSequence)
+        .setShapeDistTraveled(shapeDistTraveled)
+        .build();
+  }
+
+  public static GtfsStop createStop(
+      long csvRowNumber, String stopId, Double stopLat, Double stopLon, int locationType) {
+    return new GtfsStop.Builder()
+        .setCsvRowNumber(csvRowNumber)
+        .setStopId(stopId)
+        .setStopLat(stopLat)
+        .setStopLon(stopLon)
+        .setLocationType(locationType)
+        .build();
+  }
+
+  @Test
+  public void distanceFromToSameCoordinateIsZero() {
+    assertThat(GeospatialUtil.distanceBetweenMeter(45.508888, -73.561668, 45.508888, -73.561668))
+        .isEqualTo(0);
+  }
+
+  @Test
+  public void distanceReferenceCheck() {
+    // geographic data extracted and validated with an external tool
+    assertThat(GeospatialUtil.distanceBetweenMeter(45.508888, -73.561668, 45.507753, -73.562677))
+        .isEqualTo(148);
+  }
+
+  /**
+   * Test geospatial implementation
+   *
+   * <p>See map of trip shape and stops (in GeoJSON) at
+   * https://gist.github.com/barbeau/d9c0b90a26a3e2ba105cae5f0e8aec4a
+   *
+   * <p>For debugging, you can export a JTS-version of the buffer in WKT format using code at
+   * https://gist.github.com/barbeau/d9c0b90a26a3e2ba105cae5f0e8aec4a#gistcomment-3425554. The WKT
+   * output can then be visualized at https://arthur-e.github.io/Wicket/sandbox-gmaps3.html.
+   *
+   * <p>The spatial4j version of the buffer can't easily be visualized using GeoJSON or WKT because
+   * it uses a LineString and a proprietary "buffer" extension to GeoJSON and WKT, which most tools
+   * don't support.
+   */
+  @Test
+  public void stopWithinTripShapeBufferShouldNotGenerateNotice() {
+    NoticeContainer noticeContainer = new NoticeContainer();
+    GtfsTrip trip = createTrip(4, "route id value", "service id value", "trip id", "shape id");
+    new GtfsTrip.Builder().setTripId("trip id value").setShapeId("shape id value").build();
+    List<GtfsStopTime> stopTimes =
+        ImmutableList.of(createStopTime(5, "t1", "1001", 1), createStopTime(8, "t1", "1002", 2));
+
+    List<GtfsShape> shapePoints =
+        ImmutableList.of(
+            createShapePoint(5, "shape id", 28.05724310653972D, -82.41350776611507D, 1, 400f),
+            createShapePoint(6, "shape id", 28.05746701492806D, -82.41493135129478D, 2, 400f),
+            createShapePoint(7, "shape id", 28.05800068503469f, -82.4159394137605D, 3, 400f),
+            createShapePoint(8, "shape id", 28.05808869825447D, -82.41648754043338D, 4, 400f),
+            createShapePoint(9, "shape id", 28.05809979887893D, -82.41773971025437D, 5, 400f));
+
+    GtfsStopTableContainer stopTable =
+        GtfsStopTableContainer.forEntities(
+            ImmutableList.of(
+                createStop(2, "1001", 28.05811731042478D, -82.41616877502503D, 2),
```

Comment:
Should not the stops be of `location_type` 0 or 4?

Reply:
Indeed, thanks!